test/integration/tests/button/message.js

```diff
@@ -0,0 +1,408 @@
+                done();
+              },
+            },
+          })
+          .render("#testContainer");
+      });
+    });
+    describe("vertical layout", () => {
+      it("should place message on top by default", (done) => {
+        window.paypal
+          .Buttons({
+            style: {
+              layout: "vertical",
+            },
+            message: {},
+            test: {
+              onRender() {
+                assert.ok(getElementRecursive(".paypal-button-message-top"));
+                done();
+              },
+            },
+          })
+          .render("#testContainer");
+      });
+      it("should place message on top when position is top", (done) => {
+        window.paypal
+          .Buttons({
+            style: {
+              layout: "vertical",
+            },
+            message: {
+              position: "top",
+            },
+            test: {
+              onRender() {
+                assert.ok(getElementRecursive(".paypal-button-message-top"));
+                done();
+              },
+            },
+          })
+          .render("#testContainer");
+      });
+      it.skip("should place message on bottom when position is bottom and credit/debit is NOT a funding source", (done) => {
+        // skipped because fundingEligibility doesn't seem to be respected as passed in in this test, but confirmed to work as intended on demo page
+        window.paypal
+          .Buttons({
+            style: {
+              layout: "vertical",
+            },
+            message: {
+              position: "bottom",
+            },
+            fundingEligibility: {
+              credit: {
+                eligible: true,
+              },
+              paypal: {
+                eligible: true,
+              },
+              card: {
+                eligible: false,
+              },
+            },
+            test: {
+              onRender() {
+                assert.ok(getElementRecursive(".paypal-button-message-bottom"));
+                done();
+              },
+            },
+          })
+          .render("#testContainer");
+      });
+      it("should place message on top when no position is specified and credit/debit is a funding source", (done) => {
+        window.paypal
+          .Buttons({
+            style: {
+              layout: "vertical",
+            },
+            message: {},
+            fundingEligibility: {
+              credit: {
+                eligible: false,
+              },
+              paypal: {
+                eligible: true,
+              },
+              card: {
+                eligible: true,
+              },
+            },
+            test: {
+              onRender() {
+                assert.ok(getElementRecursive(".paypal-button-message-top"));
+                done();
+              },
+            },
+          })
+          .render("#testContainer");
+      });
```

Comment:
Similar comment as above being similar to the scenario on L253.

Reply:
true. will nix